Source/Core/Core/ScriptEngine.cpp

```diff
@@ -0,0 +1,945 @@
+// Copyright 2008 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+// ScriptEngine
+// Expands the emulator using LuaJIT scripts.
+
+#include "Core/ScriptEngine.h"
+
+#include <limits>
+#include <set>
+#include <unordered_map>
+#include <vector>
+
+#include <cassert>
+#include <lua.hpp>
+
+#include "Common/Assert.h"
+#include "Common/IniFile.h"
+
+#include "Core/ConfigManager.h"
+#include "Core/PowerPC/MMU.h"
+#include "Core/PowerPC/PowerPC.h"
+
+// clang-format off
+#define DOLPHIN_LUA_METHOD(x) {.name = #x, .func = (x)}
+// clang-format on
+
+namespace ScriptEngine
+{
+// Namespace Lua hosts static binding code.
+namespace Lua
+{
+static const char* s_context_registry_key = "c";
+static std::unordered_map<const void*, LuaFuncHandle> s_frame_hooks;
+
+// Casts a value on the Lua stack to an u32 and raises a Dolphin alert on dubious integer values.
+static std::optional<u32> luaL_checkaddress(lua_State* L, int numIdx)
+{
+  lua_Integer addr_arg = luaL_checkinteger(L, numIdx);
```

Comment:
```suggestion
  const lua_Integer addr_arg = luaL_checkinteger(L, numIdx);
```

etc in other locations for local variables.

Reply:
I believe I market most local variables as const now including pointers, I hope I didn't overdo it.